core/src/androidMain/kotlin/Scanner.kt

```diff
@@ -35,7 +35,7 @@ public class AndroidScanner internal constructor(
         ?: error("Bluetooth not supported")
 
     public override val advertisements: Flow<Advertisement> = callbackFlow {
-        val scanner = checkNotNull(bluetoothAdapter.bluetoothLeScanner) { "Bluetooth disabled." }
+        val scanner = bluetoothAdapter.bluetoothLeScanner ?: throw BluetoothDisabledException("Bluetooth disabled.")
```

Comment:
Class name already describes what went wrong. My preference would be to not be redundant with the `message` (unless we can offer more information).

```suggestion
        val scanner = bluetoothAdapter.bluetoothLeScanner ?: throw BluetoothDisabledException()
```